### PR TITLE
Added support for Section Function Attribute

### PR DESCRIPTION
**Pull request by shubhamnarlawar77**

### CODE REVIEW

**Comment by shubhamnarlawar77** [comment]
'@marxin : review request'

**Comment by marxin** [comment]
For the next time, please name branches like `csmith-gcc-add-section-attribute` ..

**Comment by shubhamnarlawar77** [comment]
I have put functions in different `usersections`. Those functions which passes `rnd_flipcoin()` will be placed in different `usersection`.

Are there any more sections? If yes, please suggest such sections.

> For the next time, please name branches like `csmith-gcc-add-section-attribute` ..

Sure, I will create new branch for new extension and delete the old one if it is fine to you?

File: src/Function.cpp [hunk 1]
```diff
@@ -133,6 +133,20 @@ AlignedFunctionAttribute::OutputAttributes(std::ostream &out)
...
+	: FunctionAttribute(name, prob)
+{
+}
+
+void
+SectionFunctionAttribute::OutputAttributes(std::ostream &out)
+{
+	if(rnd_flipcoin(FuncAttrProb)){
+		static int section = 0;
+		OutputAttribute(out, "(\"usersection" + to_string(section++) + "\")");
```
Comment (marxin) [inline src/Function.cpp]: This is fine. I would make it more random with `rnd_upto(10)` as you want to have multiple functions in a common section.

**Review by marxin — CHANGES_REQUESTED** [review]

**Comment by marxin** [comment]
> Are there any more sections? If yes, please suggest such sections.
> 
> >

I'm not aware of any special sections which we should generate.

File: src/Function.cpp (hunk 1 above)
Comment (shubhamnarlawar77) [reply src/Function.cpp]: Done.

Comment (shubhamnarlawar77) [reply src/Function.cpp]: Is pull request ready to merge?

Comment (marxin) [reply src/Function.cpp]: Yes, please merge it.

**Pull request merged**